services/worker-manager/src/data.js

```diff
@@ -46,41 +55,171 @@ WorkerType.prototype.serializable = function() {
     created: this.created.toJSON(),
     lastModified: this.lastModified.toJSON(),
     config: this.config,
-    errors: this.errors,
+    scheduledForDeletion: this.scheduledForDeletion,
     owner: this.owner,
+    wantsEmail: this.wantsEmail,
   };
 };
 
-WorkerType.prototype.reportError = async function({kind, title, description, extra, notify, owner}) {
-  await this.modify(wt => {
-    wt.errors.unshift({
-      reported: new Date(),
-      kind,
-      title,
-      description,
-      extra,
+WorkerType.prototype.compare = function(other) {
+  const fields = [
+    'name',
+    'provider',
+    'description',
+    'created',
+    'lastModified',
+    'config',
+    'scheduledForDeletion',
+    'owner',
+    'wantsEmail',
+  ];
+  return _.isEqual(_.pick(other, fields), _.pick(this, fields));
+};
+
+WorkerType.prototype.reportError = async function({kind, title, description, extra={}, notify}) {
+  if (this.wantsEmail) {
+    let extraInfo = '';
+    if (Object.keys(extra).length) {
+      extraInfo = `
+        It includes the extra information:
+
+        \`\`\`json
+        ${yaml.safeDump(extra)}
+        \`\`\`
+      `.trim();
+    }
+    await notify.email({
+      address: this.owner,
+      subject: `Taskcluster Worker Manager Error: ${title}`,
+      content: `
+  Worker Manager has encountered an error while trying to provision the workertype ${this.name}:
+
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  ${description}
+
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+  ${extraInfo}
+      `.trim(),
     });
-    wt.errors = wt.errors.slice(0, 100);
+  }
+  await WorkerTypeError.create({
+    workerType: this.name,
+    errorId: slugid.v4(),
+    reported: new Date(),
+    kind,
+    title,
+    description,
+    extra,
   });
-  await notify.email({
-    address: owner,
-    subject: `Taskcluster Worker Manager Error: ${title}`,
-    content: `
-Worker Manager has encountered an error while trying to provision the workertype ${this.name}:
-
-\`\`\`
-${description}
-\`\`\`
-
-It includes the extra information:
-
-\`\`\`json
-${yaml.safeDump(extra)}
-\`\`\`
-    `.trim(),
+};
+
+const WorkerTypeError = Entity.configure({
```

Comment:
I don't see worker errors anywhere?

Reply:
In my current design even an error for a worker would actually be a workerType error since we shouldn't be caring about individual workers in most cases.

Reply:
But I am not 100% convinced my design makes sense so we could definitely add a `WorkerErrors` table if that makes sense.